src/components/general/EntryVerify/index.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import Cloak from '#components/general/Cloak';
+import Icon from '#rscg/Icon';
+import DropdownEdit from '#components/general/DropdownEdit';
+
+import {
+    AddRequestProps,
+    Requests,
+} from '#typings';
+
+import {
+    RequestClient,
+    methods,
+} from '#request';
+import {
+    notifyOnFailure,
+    notifyOnFatal,
+} from '#utils/requestNotify';
+import _ts from '#ts';
+
+import styles from './styles.scss';
+
+interface ComponentProps {
+    id: number;
+    verified: boolean;
+    hide: (entryPermission) => boolean;
+    className?: string;
+}
+
+interface Params {
+    setVerificationStatus: (status: boolean) => void;
+    verify: boolean;
+}
+
+interface VerificationOption {
+    key: string;
+    value: string;
+    isVerified: boolean;
+}
+
+const verificationStatusOptions: VerificationOption[] = [
+    {
+        key: 'verified',
+        value: 'Verified',
+        isVerified: true,
+    },
+    {
+        key: 'unverified',
+        value: 'Unverified',
+        isVerified: false,
+    },
+];
+
+
+type Props = AddRequestProps<ComponentProps, Params>;
+
+const requestOptions: Requests<ComponentProps, Params> = {
+    setEntryVerification: {
+        url: ({ props: { id } }) => `/entries/${id}/verify/`,
+        method: methods.POST,
+        query: ({ params: { verify } }) => ({ verify }),
+        onSuccess: ({ params = {} }) => {
+            const { setVerificationStatus, verify } = params;
+            if (setVerificationStatus) {
+                setVerificationStatus(!!verify);
+            }
+        },
+        onFailure: notifyOnFailure(_ts('editEntry', 'entryVerifyFailure')),
+        onFatal: notifyOnFatal(_ts('editEntry', 'entryVerifyFailure')),
+    },
+};
+
+function EntryVerify(props: Props) {
+    const {
+        className,
+        hide,
+        verified: verifiedFromProps = false,
+        requests,
+    } = props;
+    const {
+        setEntryVerification,
+    } = requests;
+
+    const [verified, setVerificationStatus] = useState(verifiedFromProps || false);
```

Comment:
```suggestion
    const [verified, setVerificationStatus] = useState(verifiedFromProps);
```
`false` is already set as default value above